sig-cli/charter.md

```diff
@@ -0,0 +1,50 @@
+# SIG CLI Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [sig-governance].
+
+## Scope
+
+The Command Line Interface SIG (SIG CLI) is responsible for kubectl and
+related tools. This group focuses on command line tools and
+libraries to interface with Kubernetes API's.
+
+### In scope
+
+SIG CLI section in [sigs.yaml]
+
+#### Code, Binaries and Services
+
+SIG CLI code include command line tools and binaries for working with Kubernetes
+API's. Examples of these binaries include: kubectl and kustomize.
+
+### Out of scope
+
+SIG CLI is NOT responsible for defining the Kubernetes API that it
+interfaces with. The Kubernetes API is the responsibility of SIG API Machinery.
```

Comment:
It is probably worthwhile to call out all of the other CLIs that SIG CLI doesn't oversee and why. For example kubeadm, kube-apiserver, etc, etc.

Reply:
The why being that these CLIs don't interact with the API I suppose.

Reply:
There's nothing stopping us from looking after them, but nobody reached out to us for so, that's why since none of us (sig participants) authored these we don't feel like owning them nor don't want to 'steal' other's work.

Reply:
On a second note, I think that calling out what we don't oversee might be an overkill. IMO it's sufficient we clearly define ownership in the previous paragraph. The fact that API is being mentioned here is specifically to call out that we don't and never will change the API, that's sig-apimachinery responsibility.

Reply:
I am not going to hold strong on this point. But, I do still believe it would be helpful to provide illustrative counter examples.

Reply:
To recap, following Maciej's advice:

1) We won't mention kubeadm or kube-apiserver.
2) We will keep the "Out of scope" statement as is.

If misunderstood, please let me know.

Reply:
I don't think it's helpful to lump all Kubernetes components, such as kube-apiserver, under "CLI".

I do think it's helpful to indicate that SIG CLI doesn't own all conceivable command-line tools within the project: kubeadm, kops, kompose, kubefed, etc.

I would add something like:

SIG CLI is not responsible for command-line tools built and maintained by other SIGs, such as kubeadm, which is owned by SIG Cluster Lifecycle.

(Just to include one example and not make it appear to be an exhaustive list.)